### PR TITLE
[4.0] Mail template: html to plain fallback and vice versa

### PR DESCRIPTION
**Pull request by Fedik**
Pull Request for Issue #31919 .

### Summary of Changes

Mail template, html to plain fallback and vice versa.
Also I have added "scroll up" for "test email" button.

### Testing Instructions

Please follow #31919

Additionally, after send "test email" the page should be scrolled up, no need to do it manually.
So not forget to run `npm install`.

### Actual result BEFORE applying this Pull Request

The error "empty body"

### Expected result AFTER applying this Pull Request

No error, email sent.

### Documentation Changes Required
nope

### CODE REVIEW

**Comment by Fedik** [comment]
Please someone add RB tag, as it was in the original issue.

File: libraries/src/Mail/MailTemplate.php [hunk 1]
```diff
@@ -220,29 +220,41 @@ public function send()
...
-		if ($config->get('mail_style', 'plaintext') === 'html')
+		if ($mailStyle === 'plaintext' || $mailStyle === 'both')
 		{
-			$this->mailer->IsHTML(true);
-			$mail->htmlbody = $this->replaceTags(Text::_($mail->htmlbody), $this->data);
-			$this->mailer->setBody($mail->htmlbody);
+			// If the Plain template is empty try to convert the HTML template to a Plain text
+			if (!$plainBody)
+			{
+				$plainBody = strip_tags(str_replace(['<br>', '<br />'], "\n", $htmlBody));
```
Comment (PhilETaylor) [inline libraries/src/Mail/MailTemplate.php]: Maybe add `<br/>` ?

Comment (Fedik) [reply libraries/src/Mail/MailTemplate.php]: good point

Comment (Fedik) [reply libraries/src/Mail/MailTemplate.php]: done

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 2e3bf2f4efdabcd9efd46abc4d74c72fd832875d<br /><br />The email was sent to xxx@yyy.zzz using SMTP. You should check that you've received the test email.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32118">issues.joomla.org/tracker/joomla-cms/32118</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 2e3bf2f4efdabcd9efd46abc4d74c72fd832875d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32118">issues.joomla.org/tracker/joomla-cms/32118</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32118">issues.joomla.org/tracker/joomla-cms/32118</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**